food-waste-ui/src/components/Basket/Basket.jsx

```diff
@@ -0,0 +1,39 @@
+import * as React from "react";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import "./Basket.css";
+
+export default function Basket({ currentUser }) {
+  const [basket, setBasket] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const currentUserId = localStorage.getItem("current_user_id");
+
+  React.useEffect(() => {
+    setIsLoading(true);
+    axios.get("http://localhost:3001/makeapost").then((response) => {
+      let allProducts = response.data.products;
+      let allProductsInBasket = [];
+      for (let i = 0; i < allProducts.length; i++) {
+        if (allProducts[i].basket) {
+          for (let j = 0; j < allProducts[i].basket.length; j++) {
+            if (allProducts[i].basket[j] == currentUserId) {
```

Comment:
Naming is confusing, why is this checking if it == currentUserId? And what is allProducts[i].basket representing?